bootstore/src/schemes/v0/peer.rs

```diff
@@ -140,28 +167,64 @@ impl NodeHandle {
         &self,
         addrs: BTreeSet<SocketAddrV6>,
     ) -> Result<(), NodeRequestError> {
-        self.tx.send(NodeApiRequest::PeerAddresses(addrs)).await?;
+        self.tx
+            .send(NodeApiRequest::PeerAddresses(addrs))
+            .await
+            .map_err(|_| NodeRequestError::Send)?;
         Ok(())
     }
 
     /// Get the status of this node
     pub async fn get_status(&self) -> Result<Status, NodeRequestError> {
         let (tx, rx) = oneshot::channel();
-        self.tx.send(NodeApiRequest::GetStatus { responder: tx }).await?;
+        self.tx
+            .send(NodeApiRequest::GetStatus { responder: tx })
+            .await
+            .map_err(|_| NodeRequestError::Send)?;
         let res = rx.await?;
         Ok(res)
     }
 
     /// Shutdown the node's tokio tasks
     pub async fn shutdown(&self) -> Result<(), NodeRequestError> {
-        self.tx.send(NodeApiRequest::Shutdown).await?;
+        self.tx
+            .send(NodeApiRequest::Shutdown)
+            .await
+            .map_err(|_| NodeRequestError::Send)?;
         Ok(())
     }
+
+    /// Update network config needed for bringing up the control plane
```

Comment:
The fact that this passes in a `NetworkConfig` (including a `generation`) is a little weird to me.  It means that for an API consumer to update the network config, they have to first call `get_network_config` to extract its generation number, then increment it.  That seems like an internal detail that shouldn't be exposed for an "update network config" endpoint.

Reply:
This is actually critical to the correctness of the system. The local node itself is never actually updating the `NetworkConfig` directly. It's just informing the bootstore that there is an update. The update is immutable for a given `generation` and it can be written into multiple nodes simultaneously. If we did as you suggest, one node could do the RMW to update its local state simultaneously with another. They'd both bump to generation 2 but with different data. And since we only compare based on generation there would be an unknown conflict with some nodes containing some data and other nodes different data. This is because we don't have consensus or atomic updates across nodes in the bootstore. It's just a dumb replicator of versioned immutable data.

The way this is intended to be used is that update will always happen consistently in Nexus with a CTE that bumps the generation number in CockroachDB. Then a nexus instructs a sled agent in a reliable persistent workflow to inform the bootstore of the modification at the new version, and the bootstore replicates it.  

There is still a race here where the rack can crash before the update is persisted to a bootstore node, but after it is written to CockroachDB. The rack still needs to be able to boot with the old network config so it can come up and the reliable persistent workflow can run to inform the bootstore on a reboot. Even without a crash there cannot every be an atomic cutover from one network config to another because all nodes will not learn of it simultaneously. If, say, every node pointing at router a in generation 1 suddenly stopped working immediately after router b was the new thing to point at in generation 2, then there would be an outage until every node learned of the config and pointed to router b. What we want is to point at both router a and b until every node has cutover and then do another update removing router a.

Reply:
Okay, having the canonical generation number being in CRDB makes sense to me!